src/datachain/lib/file.py

```diff
@@ -190,6 +190,23 @@ def __init__(self, **kwargs):
         self._catalog = None
         self._caching_enabled: bool = False
 
+    @classmethod
+    def upload(
+        cls, path: str, data: bytes, catalog: Optional["Catalog"] = None
+    ) -> "File":
+        if catalog is None:
+            from datachain.catalog.loader import get_catalog
+
+            catalog = get_catalog()
+
+        parent = "/".join(path.split("/")[:-1])
+        name = PurePosixPath(path).name
+
```

Comment:
Is this necessary?

Reply:
Consider using `client.split_url()` in that case.

Reply:
`client.split_url` does not give what I need here as the client has already been set up with the parent path. Docs:
```
Splits the URL into two pieces:

bucket name without protocol (everything up until the first /)
path which is the rest of URL starting from bucket name
e.g s3://my-bucket/animals/dogs -> (my-bucket, animals/dogs)
```
For the test that I've written we end up writing files to `f"{src_uri}/upload-test-images/upload-test-images/image_1.jpg"`